examples/benchmarks/train_resnet_imagenet1k.py

```diff
@@ -0,0 +1,320 @@
+"""
+
+import argparse
+import logging
+import os
+
+import torch
+from torchmetrics import MetricCollection
+from torchmetrics.classification import Accuracy
+from torchvision import transforms
+from torchvision.datasets import ImageFolder
+from torchvision.models import resnet
+
+from composer import DataSpec, Time, Trainer
+from composer.algorithms import BlurPool, ChannelsLast, EMA, LabelSmoothing, ProgressiveResizing, MixUp, SAM, ColOut, RandAugment, StochasticDepth
+from composer.callbacks import SpeedMonitor, LRMonitor, CheckpointSaver
+from composer.datasets.utils import NormalizationFn, pil_image_collate
+from composer.loggers import WandBLogger
+from composer.loss import binary_cross_entropy_with_logits, soft_cross_entropy
+from composer.metrics import CrossEntropy
+from composer.models.tasks import ComposerClassifier
+from composer.optim import CosineAnnealingWithWarmupScheduler, DecoupledSGDW
+from composer.utils import dist
+
+parser = argparse.ArgumentParser()
+
+# Dataloader arguments
+parser.add_argument('data_dir', help='Path to the directory containing the ImageNet-1k dataset', type=str)
+parser.add_argument('--train_crop_size', help='Training image crop size', type=int, default=224)
+parser.add_argument('--eval_resize_size', help='Evaluation image resize size', type=int, default=256)
+parser.add_argument('--eval_crop_size', help='Evaluation image crop size', type=int, default=224)
+parser.add_argument('--train_batch_size', help='Train dataloader per-device batch size', type=int, default=2048)
+parser.add_argument('--eval_batch_size', help='Validation dataloader per-device batch size', type=int, default=2048)
+
+# Model arguments
+parser.add_argument('--model_name',
+                    help='Name of the resnet model to train',
+                    default='resnet50',
+                    choices=['resnet18', 'resnet34', 'resnet50', 'resnet101', 'resnet152'])
+parser.add_argument('--loss_name',
+                    help='Name of the loss function to use for training',
+                    default='cross_entropy',
+                    choices=['cross_entropy', 'binary_cross_entropy'])
+
+# Optimizer arguments
+parser.add_argument('--learning_rate', help='Optimizer learning rate', type=float, default=2.048)
+parser.add_argument('--momentum', help='Optimizer momentum', type=float, default=0.875)
+parser.add_argument('--weight_decay', help='Optimizer weight decay', type=float, default=5.0e-4)
+
+# LR scheduler arguments
+parser.add_argument('--t_warmup',
+                    help='Duration of learning rate warmup specified by a Time string',
+                    type=Time.from_timestring,
+                    default='8ep')
+parser.add_argument('--t_max',
+                    help='Duration to cosine decay the learning rate',
+                    type=Time.from_timestring,
+                    default='1dur')
+parser.add_argument('--alpha_f', help='Learning rate multiplier to decay to', type=Time.from_timestring, default=0)
```

Comment:
Can this arg just be the minimum learning rate to decay to? I know this framing matches the scheduler param name, but I'd rather make the top-level interface cleaner and just have a line that converts between them in the code.

Also feels ripe for a usability JIRA issue...

Reply:
Yeah, I hate this. I might just get rid of it since it doesn't need to be completely configurable and it is confusing

Reply:
Jira created